File: CS50_2017/Lecture2/Problemset2/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

string getMessage(void);
bool sanityCheck(int argumentCount,string parameter);
void encryptMessage(string msg, string key, int keyLength);
string argvToKey(string argvKey);

int main(int argc, string argv[])
{
    if (!sanityCheck(argc, argv[1]))
    {
        printf("Usage: ./vigenere k.\n");
        return 1;
    }
    
    string s = getMessage();
    int keyLength = strlen(argv[1]);
    string encryptKey = argvToKey(argv[1]);
    encryptMessage(s, encryptKey, keyLength);
}

string argvToKey(string argvKey)
{
    int n = 0;
    
    while(argvKey[n] != '\0')
    {
        if(isalpha(argvKey[n]))
        {
            argvKey[n] = toupper(argvKey[n]);
            argvKey[n] = argvKey[n] - 65;
        }
        n++;
    }
    return argvKey;
}

void encryptMessage(string msg, string key, int keyLength)
{
    int n = 0;
    int n2 = 0;
    
    while(msg[n] != '\0')
    {
        if(isalpha(msg[n]))
        {
           if(msg[n] <= 'Z' && msg[n] + key[n2 % keyLength] > 'Z')
           {
                msg[n] = msg[n] - 26;
           }
           if(msg[n] <= 'z' && msg[n] + key[n2 % keyLength] > 'z')
           {
                msg[n] = msg[n] - 26;
           }
                msg[n] = msg[n] + key[n2 % keyLength];
                n2++;
        }
        n++;
    }
    
    
    printf("ciphertext: %s\n", msg);
}

bool sanityCheck(int argumentCount, string parameter)
{
    if(argumentCount != 2)
    return false;
    int n = 0;
    while (parameter[n] != '\0')
    {
        while(!isalpha(parameter[n]))
        {
           return false;
        }
        n++;
    }
    return true;
}

string getMessage(void)
{
    printf("plaintext: ");
    string s = get_string();
    int n = 0;
    while(!isalpha(s[n]) && s[n] != ' ')
    {
        printf("Invalid input, try again: ");
        s = get_string();
        n++;
    }
    return s;
}

File: CS50_2017/Lecture8/Problemset6/vigenere.py
import sys
import cs50

def main():
    if sanityCheck(len(sys.argv), sys.argv[1]) == False:
        print("Usage: ./vigenere k.")
        return 1
    
    s = getMessage()
    keyLength = len(sys.argv[1])
    encryptKey = argvToKey(sys.argv[1])
    encryptMessage(s, encryptKey, keyLength)

def argvToKey(argvKey):
    newArgvKey = []
    for character in argvKey:
        if character.isalpha:
            newArgvKey.append(ord(character.upper()) - 65)
    return newArgvKey

def encryptMessage(msg, key, keyLength):
    ciphertext = ""
    tmptxt = 0
    n = 0;
    
    for character in msg:
        if character.isalpha:
            tmptxt = ord(character)
            if character <= 'Z' and tmptxt + key[n % keyLength] > 90:
                tmptxt -= 26;
            if character <= 'z' and tmptxt + key[n % keyLength] > 122:
                tmptxt -= 26;
            ciphertext += (chr(tmptxt + key[n % keyLength]))
            n += 1
    
    
    print("ciphertext: {}".format(ciphertext));

def sanityCheck(argumentCount, parameter):
    if argumentCount != 2:
        return False
    for i in range(len(parameter)):
        if parameter[i].isalpha == False:
           return False
    return True

def getMessage():
    print("plaintext: ", end = "")
    s = cs50.get_string()
    n = 0;
    while s[n].isalpha == False and s[n] != ' ':
        print("Invalid input, try again: ", end = "")
        s = cs50.get_string()
        n +=1
    return s

if __name__ == "__main__":
    main()
    